### ISSUE TITLE
Stuck with `X` when the dropdowns show content

### ISSUE THREAD

**Issue by outdooricon** [ptsochantaris/trailer#383]
I've just upgraded to Mojave and now i don't get numbers showing up anymore in the bar, only `X`. When i select the dropdown however, I see PR's show up. Also, I have an internet connection.

![Screen Shot 2019-03-18 at 2 16 54 PM](https://user-images.githubusercontent.com/730594/54557188-8a871800-4988-11e9-96cb-ddcc3bed2134.png)

PS - I love this app, thank you for creating it.. it saves me a lot of time!

**Comment by ptsochantaris** [comment]
Hi @outdooricon and thanks for your kind words. "X" means the last sync has failed for some reason. You can get more info by turning on "Log activity to the console" from the "Misc" tab in preferences and check what calls may be failing there. If you like, feel free to post the failing calls here and I can try to let you know what they may relate to. (Also, be sure that your API quota is good, you can see that from the "Servers" tab)

**Comment by outdooricon** [comment]
Thank you for pointing me in the right direction! I checked my api limit and it looks fine. I selected "log activity to console" but no console ever showed up for me to see. However, since you guided to it being related to sync failures, I've reset my access token and when I did that and added the new one, Trailer wiped out my watched repos list. I've noticed that if i add too many at a time, I get sync problems again. So I'm trying no to slowly add my repos again and see if I can avoid the failures..

**Comment by ptsochantaris** [comment]
Ah, sorry - the console I referred to was the system one (Console.app on macOS) - sorry for not being more specific.

Indeed, Trailer will reset all data for a server if you change the auth token, since that token could potentially be related to a different account, so it needs to sync everything "fresh" at that point, or risk corrupting all the local data.

I'm glad that things are unblocked for you - I think you're doing the right thing by enabling things incrementally. Trailer does incremental syncs on most APIs, so you can maintain a very large number of PRs and Issues listed if you like, but you can't add them all at once, as it will probably just burn through your API allowance, especially if you have enabled support for reviews, reactions, or PR CI statuses, since those APIs are specifically very wasteful due to their design, and hence Trailer does include warnings about them in the options panel :(

Trailer will warn you if your API usage is close to full and give you the time at which the API usage resets. BTW it's fine if you see a red "X" temporarily if you're close to the end of your API hour allowance, nothing will break, and Trailer will carry on when the quota is reset by GitHub. You can keep an eye on your quota use from the preferences "server" tab.

Let me know how it goes. If there is a specific repo that causes things to break, it may be worth investigating further to see if Trailer is getting confused by some data coming down from it, and at that point some console output from the API could help quite a bit.

**Comment by jakeswenson** [comment]
I am also getting these `X`'s in mojave, seeing this in the logs:
```
default	13:40:24.983715 -0700	Trailer	12.199s [C3669] path:cancel
default	13:40:24.984193 -0700	Trailer	nw_endpoint_flow_protocol_disconnected [C3669.1 IPv4#676359c1:443 cancelled socket-flow (null)] Output protocol disconnected
default	13:40:24.984328 -0700	Trailer	nw_connection_report_state_with_handler_locked [C3669] reporting state cancelled
default	13:40:24.984380 -0700	Trailer	TIC TCP Conn Destroyed [3669:0x6000012aff00]
default	13:40:25.010489 -0700	Trailer	TIC TCP Conn Event [3666:0x6000012ed200]: 2
default	13:40:25.010513 -0700	Trailer	TIC TCP Conn Cancel [3666:0x6000012ed200]
default	13:40:25.010580 -0700	Trailer	[C3666 Hostname#6ca77fec:443 tcp, url: https://<GHE_ENDPOINT>/api/v3/repos/AOrg/ARepo, tls] cancel
default	13:40:25.010756 -0700	Trailer	[C3666 Hostname#6ca77fec:443 tcp, url: https://<GHE_ENDPOINT>/api/v3/repos/AOrg/ARepo, tls] cancelled
	[C3666.1 <private> 172.24.44.198:50357<->IPv4#676359c1:443]
	Connected Path: satisfied (Path is satisfied), interface: en10, ipv4, ipv6, dns
	Duration: 15.650s, DNS @0.000s took 0.004s, TCP @0.006s took 0.038s, TLS took 0.040s
	bytes in/out: 190916/25810, packets in/out: 250/34, rtt: 0.042s, retransmitted packets: 0, out-of-order packets: 3
default	13:40:25.010799 -0700	Trailer	0.000s [C3666 <private> Hostname#6ca77fec:443 resolver] path:start
default	13:40:25.010822 -0700	Trailer	0.000s [C3666 <private> Hostname#6ca77fec:443 resolver] path:satisfied
default	13:40:25.010842 -0700	Trailer	0.000s [C3666 <private> Hostname#6ca77fec:443 resolver] resolver:start_dns
default	13:40:25.010862 -0700	Trailer	0.004s [C3666 <private> Hostname#6ca77fec:443 resolver] resolver:receive_dns
default	13:40:25.010886 -0700	Trailer	0.004s [C3666.1 <private> 172.24.44.198:50357<->IPv4#676359c1:443 socket-flow] path:start
default	13:40:25.010918 -0700	Trailer	0.004s [C3666.1 <private> 172.24.44.198:50357<->IPv4#676359c1:443 socket-flow] path:satisfied
default	13:40:25.010974 -0700	Trailer	0.006s [C3666.1 <private> 172.24.44.198:50357<->IPv4#676359c1:443 socket-flow] flow:start_connect
default	13:40:25.011079 -0700	Trailer	0.044s [C3666.1 <private> 172.24.44.198:50357<->IPv4#676359c1:443 socket-flow] flow:finish_transport
default	13:40:25.011128 -0700	Trailer	0.044s [C3666 <private> Hostname#6ca77fec:443 resolver] flow:finish_transport
default	13:40:25.011248 -0700	Trailer	0.084s [C3666.1 <private> 172.24.44.198:50357<->IPv4#676359c1:443 socket-flow] flow:finish_connect
default	13:40:25.011323 -0700	Trailer	0.084s [C3666 <private> Hostname#6ca77fec:443 resolver] flow:finish_connect
default	13:40:25.011424 -0700	Trailer	0.085s [C3666.1 <private> 172.24.44.198:50357<->IPv4#676359c1:443 socket-flow] flow:changed_viability
default	13:40:25.011471 -0700	Trailer	0.085s [C3666 <private> Hostname#6ca77fec:443 resolver] flow:changed_viability
default	13:40:25.011531 -0700	Trailer	15.650s [C3666] path:cancel
```
not sure what is going on yet. looks like the tasks were canceled... because of timeout?

**Comment by ptsochantaris** [comment]
Hi @jakeswenson - those logs are mostly system messages. What we need are more log lines that show the API activity (for instance, the "Starting refresh" line is a good one, but we need to know what happens after that)

My theory is that GitHub may have changed its API and it's breaking something, which I may need to tweak client-side. Look for something like "Wrapping up sync" and any failure messages after that, and before that you should see some API messages saying "FAILED" for some reason, with error codes.

**Comment by outdooricon** [comment]
@ptsochantaris I'll go ahead and close this. Through slowly adding my repos again, it's ben staying in sync. I also noticed that if i move a repo under a new org, I get the X until I remove that old repo. So, now that I'm aware of those two things, this issue is solved

**Comment by ptsochantaris** [comment]
Great to know @outdooricon, thanks for the update - let's keep the issue open for the moment as I'd like to see if @jakeswenson has any further feedback, I'll close it later on if there's no progress.

I'm noting the issue on moving repos between orgs - I can see how the issue could occur, and putting a TODO item to myself to see how Trailer can protect from it.

**Comment by jakeswenson** [comment]
I'll try and get more logs today,  thanks for keeping this open

**Comment by ptsochantaris** [comment]
I'll close this issue for now - @jakeswenson if you keep experiencing issues please feel free to open a new issue request.

**Comment by jakeswenson** [comment]
@ptsochantaris Sadly i have been experiencing it, but i haven't found better logs. I know @bspradling has been getting the same issues as well, maybe he has better logs?

**Comment by bspradling** [comment]
Hey @ptsochantaris, sorry to just get around to this. 

I spent some time digging through the logs and didn't see much failures other than
```
default	11:10:27.045454-0700	Trailer	Rolling back changes for failed sync on API server 'GitHub'
```

I had just readded the credential to GitHub enterprise and that was appearing to work so I tried readding my github.com credentials and that started working as well. There must have been some breaking change somewhere. 

Anyways, we are back up and running. Thanks!!
